rust/src/lib.rs

```diff
@@ -792,6 +796,64 @@ pub extern "C" fn zcashlc_get_address(
     unwrap_exc_or_null(res)
 }
 
+/// Extracts the typecodes of the receivers within the given Unified Address.
+///
+/// Returns a pointer to a slice of typecodes. `len_ret` is set to the length of the
+/// slice.
+///
+/// # Safety
+///
+/// - `ua` must be non-null and must point to a null-terminated UTF-8 string containing an
+///   encoded Unified Address.
+/// - Call [`zcashlc_free_typecodes`] to free the memory associated with the returned
+///   pointer when done using it.
+#[no_mangle]
+pub extern "C" fn zcashlc_get_typecodes_for_unified_address_receivers(
```

Comment:
we should document how the caller should map the u32's to types

Reply:
Done.